src/System.Text.Utf8/tests/RandomTests.cs

```diff
@@ -55,210 +56,107 @@ private static string GetStringLiteral(string s)
-                Assert.Equal(u8s[i], e.Current);
-                Assert.Equal((byte)s[i], (byte)(uint)cpe.Current);
-            }
-        }
-
-        [Theory]
-        [InlineData("abcdefghijklmnopqrstuvwxyz")]
-        [InlineData("ABCDEFGHIJKLMNOPQRSTUVWXYZ")]
-        [InlineData("0123456789")]
-        [InlineData(" ,.\r\n[]<>()")]
-        [InlineData("")]
-        [InlineData("1258")]
-        [InlineData("1258Hello")]
-        [InlineData("\uABCD")]
-        [InlineData("\uABEE")]
-        [InlineData("a\uABEE")]
-        [InlineData("a\uABEEa")]
-        [InlineData("a\uABEE\uABCDa")]
-        public void RoundTrip(string s)
-        {
-            Utf8String u8s = new Utf8String(s);
-            Assert.Equal(s, u8s.ToString());
-        }
-
-        [Theory]
-        [InlineData(true, "test123", "test123")]
-        [InlineData(false, "test123", "test124")]
-        public unsafe void StringEquals(bool equal, string s1, string s2)
-        {
-            byte[] b1 = Encoding.UTF8.GetBytes(s1);
-            byte[] b2 = Encoding.UTF8.GetBytes(s2);
-            Utf8String s1FromBytes = new Utf8String(b1);
-            Utf8String s2FromBytes = new Utf8String(b2);
-            fixed (byte* b1pinned = b1)
-            fixed (byte* b2pinned = b2)
-            {
-                Span<byte> sp1 = new Span<byte>(b1pinned, b1.Length);
-                Span<byte> sp2 = new Span<byte>(b2pinned, b2.Length);
-                Utf8String s1FromSpan = new Utf8String(sp1);
-                Utf8String s2FromSpan = new Utf8String(sp2);
-
-                Assert.Equal(equal, s1FromBytes == s2FromBytes);
-                Assert.Equal(equal, s1FromBytes == s2FromSpan);
-                Assert.Equal(equal, s1FromBytes == s2);
-
-                Assert.Equal(equal, s1FromSpan == s2FromBytes);
-                Assert.Equal(equal, s1FromSpan == s2FromSpan);
-                Assert.Equal(equal, s1FromSpan == s2);
-
-                Assert.Equal(equal, s1 == s2FromBytes);
-                Assert.Equal(equal, s1 == s2FromSpan);
-
-
-                Assert.Equal(equal, s1FromBytes.Equals(s2FromBytes));
-                Assert.Equal(equal, s1FromBytes.Equals(s2FromSpan));
-                Assert.Equal(equal, s1FromBytes.Equals(s2));
-
-                Assert.Equal(equal, s1FromSpan.Equals(s2FromBytes));
-                Assert.Equal(equal, s1FromSpan.Equals(s2FromSpan));
-                Assert.Equal(equal, s1FromSpan.Equals(s2));
-
-                Assert.Equal(equal, s1.EqualsUtf8String(s2FromBytes));
-                Assert.Equal(equal, s1.EqualsUtf8String(s2FromSpan));
-            }
+        public static object[][] LengthTestCases = {
+            new object[] { 0, default(Utf8String) },
+            new object[] { 0, Utf8String.Empty },
+            new object[] { 0, ""u8 },
+            new object[] { 4, "1258"u8 },
+            new object[] { 3, "\uABCD"u8 },
+            new object[] { 3, "\uABEE"u8 },
+            new object[] { 4, "a\uABEE"u8 },
```

Comment:
What is the scenario that tests for length 4 and 5 and verifying? They seem very similar to the second length 3 case.